runtime/server_http_request.go

```diff
@@ -72,12 +72,11 @@ func NewServerHTTPRequest(
 	endpoint *RouterEndpoint,
 ) *ServerHTTPRequest {
 	req := &ServerHTTPRequest{
-		httpRequest: r,
-		ctx:         r.Context(),
-		queryValues: nil,
-		metrics:     endpoint.Metrics,
-		tracer:      endpoint.tracer,
-
+		httpRequest:   r,
+		ctx:           r.Context(),
+		queryValues:   nil,
+		tracer:        endpoint.tracer,
+		metrics:       endpoint.ContextMetrics.GetOrAddInboundHTTPMetrics(r.Context()),
```

Comment:
`r.Context()` is called on line 76 in this function, let's assign a variable and only call `r.Context()` once.

Reply:
fixed